connector/gitlab/gitlab.go

```diff
@@ -269,7 +270,7 @@ func (c *gitlabConnector) groups(ctx context.Context, client *http.Client) ([]st
 		}
 
 		for _, group := range gitlabGroups {
-			groups = append(groups, group.Name)
+			groups = append(groups, strings.Replace(group.Path, "/", "-", -1))
```

Comment:
Any reason to change the separator character from slash to dash?

I suspect there is a similar fix happening on the GitLab side too so it would be good to be aligned to however GitLab is going to fix it to expose the groups directly via oidc in the future.

EDIT: Here's the GitLab MR that went in recently to add groups to the oidc claims: https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/16929/diffs#c763217d83955db9b5d8edeaf64c1fa43c8ef54b_83_95 -- they're using the group path as-is with the slash separator so I think we should do the same here.